### PR TITLE
[SPARK-48828][DOCS] Update documentation to add `column` as alias of `col`

### PR DESCRIPTION
**Pull request by thomhart31**
### What changes were proposed in this pull request?

This PR updates the `col` docstring to indicate that `col` is an alias of `column`.

### Why are the changes needed?

To improve clarity in documentation.

### Does this PR introduce _any_ user-facing change?

No changes in behavior are introduced.

### How was this patch tested?

Existing tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: c [hunk 1]
```diff
@@ -0,0 +1,2891 @@
+--global
```
Comment (allisonwang-db) [inline c:1]: @thomhart31, this seems to be the wrong file pushed :)

Comment (thomhart31) [reply c]: @allisonwang-db Good catch. Deleted

**Comment by yaooqinn** [comment]
Hi @thomhart31 The use of MINOR and JIRA ID in the PR title is mutually exclusive.

File: python/pyspark/sql/functions/builtin.py [hunk 2]
```diff
@@ -233,6 +233,7 @@ def lit(col: Any) -> Column:
 def col(col: str) -> Column:
     """
     Returns a :class:`~pyspark.sql.Column` based on the given column name.
+    An alias of :func:`column`.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions/builtin.py:236]: If you'd like to do this, you should better create the dedicated docstring for `column`, and say `column` is an alias.

**Pull request closed without merging**